munimap/static/js/digitize/digitize-controller.js

```diff
@@ -49,8 +52,11 @@ angular.module('munimapDigitize')
             // created element, i.e, it can show tabs for content that should be shown, like the attribute form,
             // even when there are no attributes present.
             $olOn(MapService.getMap(), 'singleclick', (evt) => {
-                MapService.getMap().forEachFeatureAtPixel(evt.pixel, function(feature, layer) {
-                    $scope.$parent.$parent.openDigitizePopup(layer, feature);
+                MapService.getMap().forEachFeatureAtPixel(evt.pixel, function(feature) {
+                    $scope.$parent.$parent.openDigitizePopup(feature);
+                }, {
+                    layerFilter: candidate => candidate === $scope.drawLayer.olLayer,
+                    hitTolerance: 10
                 });
```

Comment:
I don't exactly get why sometimes the function in the `digitize-popup-controller` is called and sometimes this function.

Normally you would need to filter the features to not contain the ones that are removed, but this is also done by the `featureFilter` that is setted in the `digitize-popup-controller` and used in the `draw-directive`.

Here a `forEach` is used. This is not the best way I think. If `getFeaturesAtPixel` is used and only the first feature is used, then you would need to account for the removed feature as well.

The whole popup logic is quite convoluted.